libs/api/src/lib/graphql/subscription/subscription.private-queries.ts

```diff
@@ -70,3 +71,163 @@ export const getSubscriptionsAsCSV = async (
 
   return mapSubscriptionsAsCsv(subscriptions)
 }
+
+export const getNewSubscribersYear = async (
+  authenticate: Context['authenticate'],
+  subscription: PrismaClient['subscription']
+) => {
+  const {roles} = authenticate()
+  authorise(CanGetSubscriptions, roles)
+  return (
+    await Promise.all([
```

Comment:
Can you find a cleaner way to cover the same functionality? Because it's too much code duplication imo.

Reply:
This could be done with 1 query and then counting on the server, would be less code duplication and faster as multiple DB calls are slower than manual computation generally.